### ISSUE TITLE
Self flash-read protection

### ISSUE THREAD

**Issue by szszszsz** [Nitrokey/nitrokey-start-firmware#15]
Enable flash read-protection in the firmware, right on the very first device initialization.

Related: #14

**Comment by coelner** [comment]
As reminder: https://blog.zapb.de/stm32f1-exceptional-failure/

**Comment by szszszsz** [comment]
@coelner Indeed, in this view such solution is not that much needed to implement anymore.

**Comment by coelner** [comment]
Or, like the solokey, offer a normal and a hacker version The normal version could be sealed with your glitter epoxy. Without the photo, but atleast used as visible seal?

**Comment by szszszsz** [comment]
We thought about that, but epoxy does not scale well for the production. We plan to mitigate this problem by moving to another MCU in the next hardware revision.
As for the user data, these are secure as long as the PIN is not brute-forced, which is why we ask to provide longer PINs since latest firmwares.
